### PR TITLE
Implement geographic subsetting for soil moisture dataset

### LINKED ISSUE #28: ValidationError: '5 is greater than the maximum of -180' when requesting a cube with spatial subset for several datasets
When requesting a cds cube with a spatial subset, an Validation Error occurs for some dataset ids:  `ValidationError: '5 is greater than the maximum of -180'`
this happens for the following dataset ids: 

- satellite-soil-moisture:saturation:daily
- satellite-soil-moisture:saturation:10-day
- satellite-soil-moisture:saturation:monthly
- satellite-soil-moisture:volumetric:daily
- satellite-soil-moisture:volumetric:10-day
- satellite-soil-moisture:volumetric:monthly

to reproduce: 

```
from xcube.core.store import find_data_store_extensions
from xcube.core.store import get_data_store_params_schema
from xcube.core.store import new_data_store

store = new_data_store('cds')

dataset = store.open_data('satellite-soil-moisture:saturation:daily', 
                          variable_names=['soil_moisture_saturation'], 
                          bbox=[5, 35, 6, 36], 
                          spatial_res=0.25, 
                          time_range=['1978-11-01', '1978-11-06'])
dataset
```

The stacktrace: 

``` 
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
<ipython-input-6-82c0cb0543ee> in <module>
      3                           bbox=[5, 35, 6, 36],
      4                           spatial_res=0.25,
----> 5                           time_range=['1978-11-01', '1978-11-06'])
      6 dataset

~/Desktop/projects/xcube-cds/xcube_cds/store.py in open_data(self, data_id, opener_id, **open_params)
    768         self._assert_valid_opener_id(opener_id)
    769         self._validate_data_id(data_id)
--> 770         return super().open_data(data_id, **open_params)
    771 
    772     ###########################################################################

~/Desktop/projects/xcube-cds/xcube_cds/store.py in open_data(self, data_id, **open_params)
    417 
    418         schema = self.get_open_data_params_schema(data_id)
--> 419         schema.validate_instance(open_params)
    420         handler = self._handler_registry[data_id]
    421 

~/Desktop/projects/xcube/xcube/util/jsonschema.py in validate_instance(self, instance)
    101                             format_checker=jsonschema.draft7_format_checker,
    102                             # We have to explicitly sanction tuples as arrays.
--> 103                             types=dict(array=(list, tuple)))
    104 
    105     def to_instance(self, value: Any) -> Any:

~/miniconda3/envs/xcube/lib/python3.7/site-packages/jsonschema/validators.py in validate(instance, schema, cls, *args, **kwargs)
    932     error = exceptions.best_match(validator.iter_errors(instance))
    933     if error is not None:
--> 934         raise error
    935 
    936 

ValidationError: 5 is greater than the maximum of -180

Failed validating 'maximum' in schema['properties']['bbox']['items'][0]:
    {'maximum': -180, 'minimum': -180, 'type': 'number'}

On instance['bbox'][0]:
    5
```

### PR DESCRIPTION
**Pull request by pont-us**
Fixes #28.

The only supported bbox for the soil moisture dataset was [-180, -90, 180, 90] (the entire globe), because the CDS API does not support subsetting for this dataset. With this PR, soil moisture geographic subsetting is implemented within the CDS Store itself, so smaller bboxes can be used.

### CODE REVIEW

**Comment by pont-us** [comment]
Note: I accidentally put aff834c in this branch rather than in master. I'm leaving it here, since this branch will be merged into master anyway.

File: xcube_cds/datasets/satellite_soil_moisture.py [hunk 1]
```diff
@@ -119,6 +119,14 @@ def read_file(self, dataset_name: str, cds_api_params: Dict,
         # and inspection of some downloaded files confirms it.
         ds = xr.open_mfdataset(paths, combine='by_coords')
         ds.attrs.update(self.combine_netcdf_time_limits(paths))
+
+        # Subsetting is not supported by the soil moisture dataset, so we
+        # handle it here instead.
+        if "bbox" in open_params and \
+                open_params["bbox"] != [-180, -90, 180, 90]:
+            xmin, ymin, xmax, ymax = open_params["bbox"]
+            ds = ds.sel(lon=slice(xmin, xmax), lat=slice(ymax, ymin))
```
Comment (forman) [inline xcube_cds/datasets/satellite_soil_moisture.py:128]: I fear, this code does not handle bboxes correctly whose longitude cross the anti-meridian.

File: xcube_cds/datasets/satellite_soil_moisture.py [hunk 2]
```diff
@@ -146,14 +154,11 @@ def get_open_data_params_schema(self, data_id: str) -> JsonObjectSchema:
...
-            # (i.e. minimum == maximum for every limit).
             bbox=JsonArraySchema(items=(
-                JsonNumberSchema(minimum=-180, maximum=-180),
-                JsonNumberSchema(minimum=-90, maximum=-90),
-                JsonNumberSchema(minimum=180, maximum=180),
-                JsonNumberSchema(minimum=90, maximum=90))),
+                JsonNumberSchema(minimum=-180, maximum=180, multiple_of=0.25),
+                JsonNumberSchema(minimum=-90, maximum=90, multiple_of=0.25),
+                JsonNumberSchema(minimum=-180, maximum=180, multiple_of=0.25),
+                JsonNumberSchema(minimum=-90, maximum=90, multiple_of=0.25))),
```
Comment (forman) [inline xcube_cds/datasets/satellite_soil_moisture.py:161]: This is neither handled by the gen UI nor can a user easily know that bboxes must be given as `multiple_of=0.25`. Subsetting will take care automaticlly by snapping to the 0,.25 degree grid.

```suggestion
                JsonNumberSchema(minimum=-180, maximum=180),
                JsonNumberSchema(minimum=-90, maximum=90),
                JsonNumberSchema(minimum=-180, maximum=180),
                JsonNumberSchema(minimum=-90, maximum=90))),
```

File: xcube_cds/datasets/satellite_soil_moisture.py [hunk 3]
```diff
@@ -146,14 +154,11 @@ def get_open_data_params_schema(self, data_id: str) -> JsonObjectSchema:
...
             # Like the bounding box, the spatial resolution is fixed.
             spatial_res=JsonNumberSchema(minimum=0.25,
                                          maximum=0.25,
```
Comment (forman) [inline xcube_cds/datasets/satellite_soil_moisture.py:164]: Use `const=0.25`

**Review by forman — CHANGES_REQUESTED** [review]

File: xcube_cds/datasets/satellite_soil_moisture.py (hunk 2 above)
Comment (pont-us) [reply xcube_cds/datasets/satellite_soil_moisture.py]: Fixed in commit ec8d394.

File: xcube_cds/datasets/satellite_soil_moisture.py (hunk 3 above)
Comment (pont-us) [reply xcube_cds/datasets/satellite_soil_moisture.py]: Fixed in commit ec8d394.

**Pull request merged**